File: 1768_merge_alternate.py
def mergeAlternately(self, word1: str, word2: str) -> str:
        new=""
        i=0
        while i < len(word1) and i < len(word2):
            new = new + word1[i] + word2[i]
            i +=1
        new = new + word1[i:]
        new = new+ word2[i:]
        return new

File: 1768_merge_alternate.c

char* mergeAlternately(char* word1, char* word2) {

    int i = 0;
    int j = 0;

    char* temp =
        (char*)malloc(sizeof(char) * (strlen(word1) + strlen(word2) + 1));

    while (word1[i] != '\0' && word2[i] != '\0') {
        temp[j++] = word1[i];
        temp[j++] = word2[i];
        i++;
    }

    if (word1[i] == '\0') {
        while (word2[i] != '\0') {
            temp[j++] = word2[i++];
        }
    } else {
        while (word1[i] != '\0') {
            temp[j++] = word1[i++];
        }
    }
    temp[j] = '\0';
    return temp;
}